icicle/appUtils/ntt/ntt.cu

```diff
@@ -700,9 +700,9 @@ namespace ntt {
    *  - `S` is the [scalar field](@ref scalar_t) of the curve;
    */
   extern "C" cudaError_t CONCAT_EXPAND(CURVE, InitializeDomain)(
-    curve_config::scalar_t primitive_root, device_context::DeviceContext& ctx, bool fast_twiddles_mode)
+    curve_config::scalar_t* primitive_root, device_context::DeviceContext& ctx, bool fast_twiddles_mode)
```

Comment:
Just curious: is it problematic to pass instances of concrete types into foreign function for whatever reason?

Reply:
I should double check this now that I am using arrays and not slices but it was problematic while using slices

Reply:
Hmm ok
Slices and arrays - you mean for representing scalars?